### PR TITLE
StyleCop and Jetbrains settings improvements

### PR DESCRIPTION
**Pull request by roji**
* Enable StyleCop support for R#/Rider as per https://blog.jetbrains.com/dotnet/2018/09/24/roslyn-analyzer-rulesets-stylecop-json-support-rider-2018-2/
* Copy EFCore.sln.DotSettings to the other solutions

### CODE REVIEW

File: EFCore.Cosmos.sln.DotSettings [hunk 1]
```diff
@@ -0,0 +1,194 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
```
Comment (smitpatel) [inline EFCore.Cosmos.sln.DotSettings]: - R# allows linking of existing dot settings file. So you don't need to keep full copy and things in sync.
- We explicitly decided not to have .DotSettings for anything other than EFCore.sln

Comment (roji) [reply EFCore.Cosmos.sln.DotSettings]: > R# allows linking of existing dot settings file.

Do you mean that R# somehow automatically picks up EFCore.sln.DotSettings when opening EFCore.Runtime.sln? At the very least this doesn't seem to be the case in Rider - it seems like EFCore.sln.DotSettings is ignored. Note that I opened https://youtrack.jetbrains.com/issue/RIDER-26503 for Jetbrains to consider this, but maybe I missed an existing feature?

Comment (smitpatel) [reply EFCore.Cosmos.sln.DotSettings]: This file:
https://github.com/aspnet/EntityFrameworkCore/blob/af13b52bfc390346551926ca56d5b0703c1eb0e7/EFCore.Cosmos.sln.DotSettings

But again we removed those files in 10780cd69185dea4b5b8a72a631a7010eab4f5f6

**Comment by ajcvickers** [comment]
@roji @AndriySvyryd @bricelam: @smitpatel and I discussed this briefly in triage. I have also had issues with R# and especially Rider not picking up settings with full fidelity. I would be in favor of taking these changes to make JetBrains stuff work correctly. Is there any strong pushback on this?

**Comment by roji** [comment]
> I would prefer file linking rather than copying whole content (like we had before), just to avoid issues in keeping things in sync in future.

I agree that if linking is possible it's better, otherwise any change has to be manually synced across all four DotSettings files. I know that this would work just fine on Linux (symbolic links) but don't have any experience managing links in git on Windows... If anyone can confirm that this works well we should do it, otherwise I can do a bit of experimentation to see how it is ([this issue doesn't seem to be very encouraging](https://stackoverflow.com/questions/5917249/git-symlinks-in-windows)).

Or just everyone switch to Linux? 😈

**Comment by AndriySvyryd** [comment]
@roji I think @smitpatel is referring to settings layering like https://github.com/aspnet/EntityFrameworkCore/blob/c13764e384947c96cb775ce873c213345c752816/EFCore.Cosmos.sln.DotSettings

**Comment by roji** [comment]
@AndriySvyryd @smitpatel that indeed seems to be the best way to handle this - any idea why these files were removed in the first place? 10780cd seems to have been a general Arcade/build infrastructure commit, but I can't see any mention of these Jetbrains settings files (@bricelam remember anything)?

In the meantime I updated this PR to have all solution-specific files reference EFCore.sln.DotSettings as was previously the case.

**Comment by AndriySvyryd** [comment]
@roji They were considered clutter because *nobody* was using ReSharper at the time.

**Comment by roji** [comment]
@AndriySvyryd ok I see, thanks!

**Pull request merged**